### DISCUSSION TITLE
Question: Markdoc->Markdown renderer?

### DISCUSSION THREAD

**Discussion by abirmingham** [markdoc/markdoc#275] [Q&A] [answered]
Greetings,

My company is considering adoption of markdoc. We would primarily use markdoc with the react renderer. However, we also have a case where where a python component deals with markdown. As markdoc does not currently have a python renderer, we are considering rendering markdoc->markdown and sending markdown to this component. I could not find a markdoc->markdown renderer, however—only html and react.

Does a markdoc->markdown renderer exist? Are there any plans to produce such a renderer?

Thank you!

**Comment by rpaul-stripe** [comment, marked as answer]
There is an experimental formatter (`Markdoc. __EXPERIMENTAL__format`) that can convert a Markdoc AST back into source. You could do transformations on the AST to remove Markdoc-specific elements and then use it to convert the content back to plain Markdown.

Depending on your use case, you could also consider serializing either the Markdoc AST or render tree to JSON to use them directly in Python.

**Reply by abirmingham** [reply]
Got it. Thank you for the quick response Ryan! Are there any plans to write a renderer to markdown, and/or any plans to implement Markdoc in any other languages such as python?

**Reply by rpaul-stripe** [reply]
We don't have plans for either of those things at this time.
